src/per/spi.cpp

```diff
@@ -341,19 +335,14 @@ SpiHandle::Result checkPinMatchSpi(GPIO_InitTypeDef* init, Pin pin, int p_num)
 {
     for(int i = 0; i < 3; i++)
     {
-        Pin tmp_pin;
-        tmp_pin.Init(pins_periphs_spi[p_num][i].port,
-                     pins_periphs_spi[p_num][i].pin);
-
         Pin none_pin;
-        none_pin.Init(pins_none_spi.port, pins_none_spi.pin);
 
-        if(tmp_pin == none_pin)
+        if(pins_periphs_spi[p_num][i].pin == none_pin)
```

Comment:
Have you tried to implement `Pin::invalid`? It would make this comparison a little prettier by removing the need for `none_pin`
```cpp
struct Pin 
{
    // ...
    static constexpr Pin invalid = { Port::DSY_GPIOX, 0 };
}
```